src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RedundantThisCheck.java

```diff
@@ -0,0 +1,386 @@
+ *     return this.x++; // OK, 'this' is necessary here
+ *   }
+ *
+ *   public void methodThree() {}
+ *
+ *   public int methodFour() {
+ *     int x = 5;
+ *     return this.x + x ; // OK, 'this' is necessary here
+ *   }
+ *
+ *   private static int y;
+ *
+ *   public void methodFive() {
+ *     this.y = 0; // violation, 'this' is not necessary here
+ *   }
+ * }
+ * </pre>
+ *
+ * <p>
+ * To configure to check the {@code this} qualifier for fields only:
+ * </p>
+ * <pre>
+ * &lt;module name="RedundantThis"&gt;
+ *   &lt;property name=&quot;checkMethods&quot; value=&quot;false&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class Foo {
+ *   private int x;
+ *
+ *   public Foo() {
+ *     this.x++; // violation, 'this' is not necessary here
+ *   }
+ *
+ *   public int methodOne(int x) {
+ *     this.methodTwo(); // OK, methods are not checked
+ *     return this.x++; // OK, 'this' is necessary here
+ *   }
+ *
+ *   public void methodTwo() {}
+ * }
+ * </pre>
+ * <p>
+ * To configure to check the {@code this} qualifier for methods only:
+ * </p>
+ * <pre>
+ * &lt;module name="RedundantThis"&gt;
+ *   &lt;property name=&quot;checkFields&quot; value=&quot;false&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class Foo {
+ *   private int x;
+ *
+ *   public Foo() {
+ *     this.x++; // OK, fields are not checked
+ *   }
+ *
+ *   public int methodOne() {
+ *     methodThree();
+ *     return this.x++; // OK, fields are not checked
+ *   }
+ *
+ *   public int methodTwo() {
+ *     this.methodThree(); // violation, 'this' is not necessary here
+ *   }
+ *
+ *   public int methodThree() {}
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code redundant.this.method}
+ * </li>
+ * <li>
+ * {@code redundant.this.variable}
+ * </li>
+ * </ul>
+ *
+ * @since 8.32
+ */
+// -@cs[ClassDataAbstractionCoupling] This check requires to work with and identify many frames.
+public class RedundantThisCheck extends AbstractFrameCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_METHOD = "redundant.this.method";
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_VARIABLE = "redundant.this.variable";
+
+    /** Control whether to check references to fields. */
+    private boolean checkFields = true;
+
+    /** Control whether to check references to methods. */
+    private boolean checkMethods = true;
+
+    /**
+     * Setter to control whether to check references to fields.
+     *
+     * @param checkFields should we check methods usage or not.
+     */
+    public void setCheckFields(boolean checkFields) {
+        this.checkFields = checkFields;
```

Comment:
@rnveach I am not able to find the code coverage issue for this line.

Reply:
So do you have a test that uses the property `checkFields` as part of it's validation?

Reply:
Yes Sir

Reply:
I see no mention of check fields. Did you confirm with the config inside in the input file?

Reply:
I am sorry, It's my bad.
Those are for different tests.
No, I don't have a test that used `checkFields`, I am only using it in JavaDoc

Reply:
But I am using it in `XpathRegressionRedundantThisTest`

Reply:
You must have a test using all its possible values outside of regression in the normal tests. This is why coverage is missing.

Reply:
But I am getting a coverage issue of setters and I don't really think that I am using them anywhere so I can remove the setters, What are your views on that?

Reply:
No. Setters are how properties work in the configuration. If there is no setter, you can not have properties for your check.

Again, you must have inputs that set TRUE and FALSE for your boolean properties. Until you do that, coverage will be missing.

Reply:
I think I should push my code to see the actual code coverage report.

Reply:
Tried my level best to deal with `checkFields `and `checklMethods`
Kindly help me to resolve it

Reply:
Create an input file and test where you set `checkFields` to `false`.

Reply:
Done!

Reply:
Link to latest report. If it is missing code coverage, you will need to figure out what you need to do to hit that code.

Point out what class/method is the one you are looking at,

Reply:
I added a test for frame check but it is not changing the Code coverage.

Test I added:
   ```
 @Test
    public void testUnusedMethod() throws Exception {
        final DetailAstImpl ident = new DetailAstImpl();
        ident.setText("testName");

        final Class<?> cls = Class.forName(AbstractFrameCheck.class.getName() + "$CatchFrame");
        final Constructor<?> constructor = cls.getDeclaredConstructors()[0];
        constructor.setAccessible(true);
        final Object o = constructor.newInstance(null, ident);

        final DetailAstImpl actual = TestUtil.invokeMethod(o, "getFrameNameIdent");
        assertSame(ident, actual, "expected ident token");
        assertEquals("CATCH_FRAME",
            TestUtil.invokeMethod(o, "getType").toString(),
                "expected catch frame type");
    }
```

Reply:
You would need to identify if there is someway to validate the code without the use of reflections. If it is not, why is the code needed? If the code is still needed a javadoc is needed for the test to say why reflection is needed.

Reply:
Is this too a DisAllowed Import: `import java.lang.reflect.Constructor;`

Reply:
Tried a lot to handle code coverage for Frames but got a code coverage issue, not really getting the actual point.

Reply:
@rnveach I am getting  DisAllowed Import for this too:  `import java.lang.reflect.Constructor;`

Reply:
> If the code is still needed a javadoc is needed for the test to say why reflection is needed.

In addition, you would need to suppress the violation as this is why it is there, to ask the additional questions on why reflection is needed and more documentation to say why.

Reply:
@rnveach  Do RedundantThis use a `validateOnlyOverlapping` set?

Reply:
@Rahulkhinchi03 if the `getFrameNameIdent` method is never called, you can remove this method. Then run the regression tests. You will either get confirmation that the method is not needed, or you will get an example of the source code that needs it.